types/coin.go

```diff
@@ -311,7 +311,7 @@ func (coins Coins) IsAnyGTE(coinsB Coins) bool {
 
 	for _, coin := range coins {
 		amt := coinsB.AmountOf(coin.Denom)
-		if coin.Amount.GTE(amt) {
+		if coin.Amount.GTE(amt) && !amt.IsZero() {
```

Comment:
Wouldn't it be preferable to have `AmountOf` signal that the denomination was not present among the coins it held, e.g. by returning a second, boolean value? As it stands, there is no distinction between "balance of zero" and "this denomination was not present", which all callers of `AmountOf` should keep in mind.

Reply:
Possibly, but in a way `AmountOf` already does this. Assuming a valid coin set should never contain zero coins, `AmountOf` will always return a positive amount OR zero if the denom doesn't exist.

Reply:
True enough, although recent events would indicate that not all consumers of the function are aware of this distinction. Introducing a specific return value for that case would be helpful towards forcing callers to take that possibility into account.

Reply:
@haasted yes! We really need to make clear that coins should always be sorted and that they should never include zero amounts (which existing APIs provide).